### ISSUE TITLE
Cannot add Indexers from Jackett

### ISSUE THREAD

**Issue by misleadingrhino** [Radarr/Radarr#236]
So far, great software. 

However, I cannot add an indexer from Jackett using the same configuration as Sonarr. Below is the error I get:

> Unable to connect to indexer: Sequence contains no elements

> System.InvalidOperationException: Sequence contains no elements
>   at System.Linq.Enumerable.First[TSource] (IEnumerable`1 source) <0x403605c0 + 0x0013f> in <filename unknown>:0 
>   at NzbDrone.Core.Indexers.HttpIndexerBase`1[TSettings].TestConnection () [0x0000e] in C:\projects\radarr-usby1\src\NzbDrone.Core\Indexers\HttpIndexerBase.cs:323

Everything works great using the built-in trackers. 

I am using the docker image from linuxserver.io (x64).

Thanks!

**Comment by galli-leo** [comment]
Please add anything to the username field.

**Comment by misleadingrhino** [comment]
This was using Torznab(which has no username field). Is TorrentPotato the preferred format?

**Comment by onedr0p** [comment]
yes use torrentpotato, @galli-leo should we remove Torznab?

**Comment by hotio** [comment]
What are the benefits of TorrentPotato?
Torznab has 1 extra feature "Categories filtering" as far as I can tell.

**Comment by onedr0p** [comment]
I'm not sure of the difference between the two, torrentpotato is made to work specifically with movies. We can add that to torrentpotato, if you want, add a GH issue and i'll take a look.

**Comment by hotio** [comment]
I'm not really sure which implementation is the best torrentpotato vs torznab...

But I've got this thing in my head... that if tornzab works 100% OK, we get this...

Sonarr:
-Newznab
-Torznab

Radarr:
-Newznab
-Torznab

Why the need for TorrentPotato (invented if I recall correct by CouchPotato, which if Radarr continues at this pace, will be obsolete by February)?

Unless TorrentPotato has some major advantage...

**Comment by onedr0p** [comment]
They are the same but have different ways of getting to the information (one uses JSON, the other uses XML). TorrentPotato has become a standard agnostic to couchpotato.

**Comment by kaso17** [comment]
Hi, I would have no problem adding movie-search to Jackett.
It probably would be a more specific version of the default search with the cat argument set to 2000 (Movies) by default.
Additionally I wouldn't set supportedParams=imdbid by default. Most indexers don't support direct searching by IMDB id. In this case Jackett would have to lookup the movie name itself (again). It would be just a useless imdb API call. So Radarr should use imdbid if it's signaled as supported argument of the indexer and otherwise fallback to simple movie title based (q) search, that way jackett doesn't have to redo the IMDB API call.

**Comment by galli-leo** [comment]
@kaso17 Actually we are doing that right now. If the imdbid caps is there we search for imdbid, else using the query. So if you could just add the imdbid to the caps (if it's supported) that would be very cool!

**Comment by kaso17** [comment]
Just pushed a new jackett pre-release (v0.7.884) containing the potato feed change + movie-search with imdbid support for the torznab endpoint.
Currently only PTP, TorrentDay, TehConnection and IPTorrent support direct imdbid search.
imdbid is only supported for movie-search, if imdbid is specified you must not specify q.
If imdbid is used there's no filtering by category. If q is used it filters by movie categories by default.
In both cases category filtering can be overridden by specifying the cat argument (empty string means no filtering)
Please give it a try.

**Comment by eephyne** [comment]
I posting here because my issue seem to be the same as the OP.
I downloaded the last release of jackett and added it to radarr, it work for some research but I didn't get the result expected (seem to be related to the indexer). But then I changed some settings like the parsing or the category, then nothing worked.
I removed the indexer from radarr and tried to re-add it but I can no longer, it say 
```
 [v0.2.0.870] System.InvalidOperationException: Sequence contains no elements
  at System.Linq.Enumerable.First[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00010] in <a16ef142877c4302b96a103db87ba16b>:0 
  at NzbDrone.Core.Indexers.HttpIndexerBase`1[TSettings].TestConnection () [0x0001a] in <9ad4852ba2664806bd2189b6172b8818>:0 
```
I upgraded Radarr but still the same error

I precise that the exact same settings work on sonarr.

I really think this is a radarr issue and not a jackett one because 1/ it work well with sonarr and 2/ when I shut off jackett, radarr throw the exact same message. (considering the message, I don't think it start the request in the first place anyway)

**Comment by kaso17** [comment]
@eephyne please provide the torznab request URL (from the sonarr logs).

**Comment by jeff15110168** [comment]
I'm having the same issue trying to connect BTDB to Radarr (see Radarr logs below):

> 18-2-24 18:09:08.2|Warn|Torznab|Unable to connect to indexer
> 
> [v0.2.0.980] System.InvalidOperationException: Sequence contains no elements
>    at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
>    at NzbDrone.Core.Indexers.HttpIndexerBase`1.TestConnection() in C:\projects\radarr-usby1\src\NzbDrone.Core\Indexers\HttpIndexerBase.cs:line 324
> 
> 18-2-24 18:09:08.2|Warn|NzbDroneErrorPipeline|Invalid request Validation failed: 
>  -- Unable to connect to indexer, check the log for more details
> 

It works fine in Sonarr (Sonarr log below):

> 18-2-24 18:30:13.6|Debug|Torznab|Downloading Feed http://[ADDRESS]/api/v2.0/indexers/btdb/results/torznab/api?t=tvsearch&cat=100001&extended=1&apikey=(removed)&offset=0&limit=100
> 18-2-24 18:30:17.0|Debug|Api|[POST] /api/indexer/test: 200.OK (3403 ms)

**Comment by galli-leo** [comment]
@jeff15110168 Disable Search, add indexer, enable search. This is because „officially“ your indexer does not support movie searching

**Comment by jeff15110168** [comment]
I disabled the search button when trying to add, but it still does not add (test fails)

**Comment by galli-leo** [comment]
@jeff15110168 Tgen try disabling RSS too

**Comment by jeff15110168** [comment]
That didn’t work either :/

On Feb 25, 2018, at 4:58 AM, Leonardo Galli <notifications@github.com<mailto:notifications@github.com>> wrote:


@jeff15110168<https://github.com/jeff15110168> Tgen try disabling RSS too

**Comment by galli-leo** [comment]
Please post the full log

**Comment by stroi08** [comment]
First post, so - Thank you for the effort with Radarr, much apprecoated!

I'm also having problems adding Jackett indexes. Tried the suggestions in this thread without luck. Log is below. Any suggestions?
I noticed Jackett is at http://127.0.0.1:9117, while I was expecting my PC's (192.168.0.16), in case that can be the cause.


NzbDroneErrorPipeline:
> Invalid request Validation failed: 
 -- Unable to connect to indexer, check the log for more details

Torznab:
> Unable to connect to indexer: HTTP request failed: [500:InternalServerError] [GET] at [http://127.0.0.1:9117/api/v2.0/indexers/zooqle/results/torznab/api?t=movie&cat=2000&extended=1&apikey=t5j7cq7kvssg4uww1ynt9s0bohmnsj4i&offset=0&limit=100]

HTTPClient:
> HTTP Error - Res: [GET] http://127.0.0.1:9117/api/v2.0/indexers/zooqle/results/torznab/api?t=movie&cat=2000&extended=1&apikey=(removed)&offset=0&limit=100: 500.InternalServerError
{
  "result": "error",
  "error": "Exception (zooqle): Fjernserveren returnerede en fejl: (400) Forkert anmodning.: Der opstod en fejl under afsendelse af anmodningen.",
  "stacktrace": "   ved Jackett.Common.Indexers.BaseIndexer.<ResultsForQuery>d__64.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved Jackett.Common.Indexers.BaseWebIndexer.<ResultsForQuery>d__25.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)\r\n   ved Jackett.Controllers.ResultsController.<Torznab>d__14.MoveNext() i C:\\projects\\jackett\\src\\Jackett\\Controllers\\ResultsController.cs:linje 312\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved System.Threading.Tasks.TaskHelpersExtensions.<CastToObject>d__1`1.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved System.Web.Http.Tracing.ITraceWriterExtensions.<TraceBeginEndAsyncCore>d__17`1.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved System.Web.Http.Controllers.ApiControllerActionInvoker.<InvokeActionAsyncCore>d__1.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved System.Web.Http.Tracing.ITraceWriterExtensions.<TraceBeginEndAsyncCore>d__17`1.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved System.Web.Http.Filters.ActionFilterAttribute.<ExecuteActionFilterAsyncCore>d__5.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved System.Web.Http.Filters.ActionFilterAttribute.<ExecuteActionFilterAsyncCore>d__5.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved System.Web.Http.Controllers.ActionFilterResult.<ExecuteAsync>d__5.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved System.Web.Http.Filters.AuthorizationFilterAttribute.<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved System.Web.Http.Controllers.ExceptionFilterResult.<ExecuteAsync>d__6.MoveNext()",
  "innerstacktrace": "   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved CloudFlareUtilities.ClearanceHandler.<SendAsync>d__20.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)\r\n   ved Jackett.Common.Utils.Clients.HttpWebClient.<Run>d__7.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)\r\n   ved Jackett.Common.Utils.Clients.WebClient.<GetString>d__18.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved Jackett.Common.Indexers.BaseWebIndexer.<RequestStringWithCookies>d__5.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)\r\n   ved Jackett.Common.Indexers.CardigannIndexer.<PerformQuery>d__29.MoveNext()\r\n--- Afslutningen på staksporingen fra den tidligere placering, hvor undtagelsen blev udløst ---\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   ved System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   ved Jackett.Common.Indexers.BaseIndexer.<ResultsForQuery>d__64.MoveNext()"
}
